minari/utils.py

```diff
@@ -394,10 +397,17 @@ def create_dataset_from_buffers(
     except InvalidSpecifier:
         print(f"{minari_version} is not a version specifier.")
 
+    if env is None and (observation_space is None or action_space is None):
+        raise ValueError(
+            "Both observation space and action space must be provided, if environment is None"
+        )
+
     if observation_space is None:
-        observation_space = env.observation_space
+        observation_space = (
+            env.observation_space  # pyright: ignore[reportOptionalMemberAccess]
+        )
     if action_space is None:
-        action_space = env.action_space
+        action_space = env.action_space  # pyright: ignore[reportOptionalMemberAccess]
```

Comment:
same as above

Reply:
Yes , pyright error for optional member access.